static/js/browser2/weather_chart_section.tsx

```diff
@@ -90,26 +96,35 @@ export class WeatherChartSection extends React.Component<
         .get(`/weather?dcid=${this.props.dcid}&prop=${prop}`)
         .then((resp) => resp.data);
     });
-    Promise.all(weatherPromises).then((weatherPromisesData) => {
-      const propToSourceSeries = {};
-      weatherPromisesData.forEach((weatherData) => {
-        if (_.isEmpty(weatherData)) {
-          return;
-        }
-        const values = {};
-        weatherData.forEach((data) => {
-          values[data.observationDate] = data.meanValue;
+    loadSpinner(LOADING_CONTAINER_ID);
+    Promise.all(weatherPromises)
+      .then((weatherPromisesData) => {
+        const propToSourceSeries = {};
+        weatherPromisesData.forEach((weatherData) => {
+          if (_.isEmpty(weatherData)) {
+            return;
```

Comment:
removeSpinner() before return?

Reply:
return in a forEach is actually equivalent to continue